src/pages/HomePage.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { fetchCoinsAction } from '../redux/slices/coinSlice';
+
+import CoinComponent from '../components/Coin';
+import '../styles/HomePage.css';
+
+const HomePage = () => {
+  const dispatch = useDispatch();
+  const coins = useSelector((state) => state.coins.coins);
+
+  useEffect(() => {
+    dispatch(fetchCoinsAction());
+  }, [dispatch]);
+
+  const [search, setSearch] = useState('');
+
+  const handleChange = (e) => {
+    setSearch(e.target.value);
+  };
+
+  const filteredCoins = coins.filter((coin) => coin.name.toLowerCase().includes(search.toLowerCase()));
+
+  return (
+    <div className="home-page">
+      <div className="searchBar">
+        <input
+          className="search"
+          type="text"
+          placeholder="Search for Crypto Coin"
+          onChange={handleChange}
+        />
+      </div>
```

Comment:
- Currently, your search doesn't seem to work as expected, when enter any coin name, it doesn't filter it out. please, double-check this to fix it as expected. if this is an oversight from my end, you can ignore this. but currently, it doesn't seem to work.

![image](https://github.com/Peter-Kibet/metrics-app-react-capstone/assets/59397159/8c4ec794-27cf-4b20-bccf-61e6b8ffb3d9)